From e8a284c499e15e3b2b8ff21117b5fa832d0a044b Mon Sep 17 00:00:00 2001
From: Simon McVittie <smcv@debian.org>
Date: Thu, 26 Oct 2017 16:02:07 +0100
Subject: [PATCH 3/3] GtkWindow: Set tiled class on windows that have any tiled
 edge

This has no practical effect on Adwaita, but might fix some
third-party themes that were broken by GTK+ 3.22.23.

Signed-off-by: Simon McVittie <smcv@debian.org>
Applied-upstream: 3.22.25, commit:72045a1a2defa82692bac043c5123da7944d76ec
---
 gtk/gtkwindow.c | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/gtk/gtkwindow.c b/gtk/gtkwindow.c
index a4ba098224..eb3b33578a 100644
--- a/gtk/gtkwindow.c
+++ b/gtk/gtkwindow.c
@@ -7636,14 +7636,12 @@ update_window_style_classes (GtkWindow *window)
 
   context = gtk_widget_get_style_context (GTK_WIDGET (window));
 
-  if (priv->edge_constraints == 0)
-    {
-      if (priv->tiled)
-        gtk_style_context_add_class (context, "tiled");
-      else
-        gtk_style_context_remove_class (context, "tiled");
-    }
+  if (priv->tiled)
+    gtk_style_context_add_class (context, "tiled");
   else
+    gtk_style_context_remove_class (context, "tiled");
+
+  if (priv->edge_constraints != 0)
     {
       guint edge_constraints = priv->edge_constraints;
 
-- 
2.15.0.rc2

